delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/typehierarchy/model/TypeHierarchyRejection.scala

```diff
@@ -0,0 +1,48 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.typehierarchy.model
+
+import ch.epfl.bluebrain.nexus.delta.kernel.error.Rejection
+
+sealed abstract class TypeHierarchyRejection(val reason: String) extends Rejection
+
+object TypeHierarchyRejection {
+
+  /**
+    * Enumeration of possible reasons why type hierarchy is not found
+    */
+  sealed abstract class NotFound(reason: String) extends TypeHierarchyRejection(reason)
+
+  /**
+    * Signals that the type hierarchy does not exist.
+    */
+  final case object TypeHierarchyDoesNotExist extends NotFound("The type hierarchy does not exist.")
+
+  /**
+    * Signals an attempt to retrieve a type hierarchy at a specific revision when the provided revision does not exist.
+    *
+    * @param provided
+    *   the provided revision
+    * @param current
+    *   the last known revision
+    */
+  final case class RevisionNotFound(provided: Int, current: Int)
+      extends NotFound(s"Revision requested '$provided' not found, last known revision is '$current'.")
+
+  /**
+    * Signals that the provided revision does not match the latest revision
+    *
+    * @param provided
+    *   provided revision
+    * @param expected
+    *   latest know revision
+    */
+  final case class IncorrectRev(provided: Int, expected: Int)
```

Comment:
We could drop this in this case as the risk of concurrent updates will be very low.

Reply:
Ok but what do we do with the rejection that needs to be provided if a unique violation happens? It can't be a partial match there. Just some generic rejection, assuming it won't happen?

Reply:
Sorry, I forgot about this one :).
What I meant is that we don't have to handle it in the api endpoint and in the state machine

Reply:
Done 👍